axelor-base/src/main/java/com/axelor/apps/base/service/LanguageServiceImpl.java

```diff
@@ -0,0 +1,197 @@
+package com.axelor.apps.base.service;
+
+import com.axelor.apps.base.db.Language;
+import com.axelor.apps.base.db.repo.LanguageRepository;
+import com.axelor.db.JPA;
+import com.axelor.db.Query;
+import com.axelor.meta.db.MetaSelect;
+import com.axelor.meta.db.MetaSelectItem;
+import com.axelor.meta.db.repo.MetaSelectItemRepository;
+import com.axelor.meta.db.repo.MetaSelectRepository;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.util.List;
+
+public class LanguageServiceImpl implements LanguageService {
+  protected MetaSelectRepository metaSelectRepository;
+  protected MetaSelectItemRepository metaSelectItemRepository;
+  protected LanguageRepository languageRepository;
+
+  private static final String SELECT_NAME = "select.language";
+  private static final String NATIVE_SELECT_LANGUAGE_MODULE = "axelor-core";
+  private static final String NEW_SELECT_LANGUAGE_MODULE = "axelor-base";
+
+  @Inject
+  public LanguageServiceImpl(
+      MetaSelectRepository metaSelectRepository,
+      MetaSelectItemRepository metaSelectItemRepository,
+      LanguageRepository languageRepository) {
+    this.metaSelectRepository = metaSelectRepository;
+    this.metaSelectItemRepository = metaSelectItemRepository;
+    this.languageRepository = languageRepository;
+  }
+
+  /**
+   * If we are modifying an existing object, pass 'false' to isNew. Otherwise, if we want to save a
+   * new Language obj, pass 'true' to isNew, and pass 'null' to oldName and oldCode.
+   *
+   * @param language
+   * @param oldName
+   * @param oldCode
+   * @param isNew
+   */
+  @Transactional
+  @Override
+  public void saveLanguageToMetaSelect(
+      Language language, String oldName, String oldCode, boolean isNew) {
+    // find the overridden metaSelect object in axelor-base module
+    Query<MetaSelect> metaSelectQuery =
+        metaSelectRepository
+            .all()
+            .filter("self.name = :selectName and self.module = :newModuleName");
+    metaSelectQuery
+        .bind("selectName", SELECT_NAME)
+        .bind("newModuleName", NEW_SELECT_LANGUAGE_MODULE);
+    MetaSelect metaSelect = metaSelectQuery.fetchOne();
+    if (metaSelect == null) {
+      // A new MetaSelect should be created.
+      metaSelect = copyMetaSelect();
+      if (metaSelect == null) return;
+    }
```

Comment:
You only need to find the metaSelect if you're going to add a new MetaSelectItem, so if isNew = true
Else if isNew != true then the metaSelect is just `metaSelectItem.getMetaSelect()`